orchestrator/careplanservice/handle_get.go

```diff
@@ -0,0 +1,191 @@
+package careplanservice
+
+import (
+	"context"
+	"encoding/json"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+	"strings"
+)
+
+func (s *Service) handleGet(httpResponse http.ResponseWriter, request *http.Request) error {
+	queryParams := request.URL.Query()
+	urlParts := strings.Split(strings.TrimPrefix(request.URL.Path, basePath+"/"), "/")
+	switch urlParts[0] {
+	case "CarePlan":
+		// fetch CarePlan + CareTeam, validate requester is participant of CareTeam
+		// if there are query params present, we need to handle the request a bit differently
+		if len(queryParams) != 0 {
+			if queryParams.Get("_id") == "" {
+				return coolfhir.NewErrorWithCode("Query parameter _id is not present, can't find CarePlan", http.StatusBadRequest)
+			}
+
+			var bundle fhir.Bundle
+			err := s.fhirClient.Read("CarePlan", &bundle, fhirclient.QueryParam("_id", queryParams.Get("_id")), fhirclient.QueryParam("_include", "CarePlan:care-team"))
+			if err != nil {
+				return err
+			}
+			var careTeams []fhir.CareTeam
+			err = coolfhir.ResourcesInBundle(&bundle, coolfhir.EntryIsOfType("CareTeam"), &careTeams)
+			if err != nil {
+				return err
+			}
+			if len(careTeams) == 0 {
+				return coolfhir.NewErrorWithCode("CareTeam not found in bundle", http.StatusNotFound)
+			}
+
+			err = validatePrincipalInCareTeams(request.Context(), careTeams)
+			if err != nil {
+				return err
+			}
+
+			// Valid requester, now perform the bundle request again with all the params
+			params := []fhirclient.Option{}
```

Comment:
```suggestion
			// TODO: Check if all parameters are allowed
			params := []fhirclient.Option{}
```
@bramwesselo ?

Reply:
well, I'm trying to make sense of this code. I see that there's a fhir-client-repo, some coolfhir-lib and a zorgbijjou-fhir-model-fork; it's a lot to get my head around...

Reply:
I have pushed this out in distinct commits as best I could, might be easier to review changes per commit